pilot/pkg/model/sidecar.go

```diff
@@ -413,17 +464,35 @@ func (ilw *IstioEgressListenerWrapper) VirtualServices() []Config {
 	return ilw.virtualServices
 }
 
-// DependsOnNamespace determines if the Sidecar includes the given namespace.
-func (sc *SidecarScope) DependsOnNamespace(namespace string) bool {
+// DependsOnConfig determines if the proxy depends on the given config.
+// Returns whether depends on this config and whether this kind of config is scoped(or known to be depended or not) here.
+func (sc *SidecarScope) DependsOnConfig(config ConfigKey) (bool, bool) {
```

Comment:
I strongly suggest returning true or false. For what I do not know, return true, leaves it to the other decision-makers. Keep in mind the decision is made in order, not once for all.

Reply:
If it returns true for "unknown" case, other decision-makers could do nothing later as the "unknown" info is lost. 
I just think it's not the sidecarScope's responsibility to consider unknown as "depends-on".

Reply:
@howardjohn What do u think of this?

Reply:
>  If it returns true for "unknown" case, other decision-makers could do nothing later as the "unknown" info is lost. I just think it's not the sidecarScope's responsibility to consider unknown as "depends-on".

I see your point about interpreting unknown as "true" which the end result might incorrectly push, but I do think returing two bools is confusing

How about changing it as using `DependencyResult` or some other enum?
`func (sc *SidecarScope) DependsOnConfig(config ConfigKey) DependencyResult`

and we can return `Yes`, `No`, `Unknown` - so the code is more readable? This scoped is, IMO confusing

Reply:
I agree its confusing. Personal preference is to return a single bool. In the current code, this will remain identical behavior. If we need to extend the functionality later, we can use `DependencyResult` or similar. no need to make the code more complex now than it needs to be

Reply:
OK. I've change that.
I just referred the index expression with multiple return value which also tells not only the return value itself but also "existence status" to avoid ambiguity.